File: java/src/main/java/fr/arolla/Booking.java
package fr.arolla;

public class Booking {
  private final Inventory inventory;
  private final Ticketing ticketing;

  public Booking(Inventory inventory, Ticketing ticketing) {
    this.inventory = inventory;
    this.ticketing = ticketing;
  }

  public void book(int numSeats) {
    System.out.format("Booking %d seats", numSeats);
    inventory.decrementCapacity(numSeats);
    ticketing.printTicket(numSeats);
  }
}

File: dotnet/Choreography/Booking.cs
using Choreography.Events;

namespace Choreography;

public class Booking
{
    private readonly Context _context;

    public Booking(Context context)
    {
        _context = context;
    }

    public void Book(int numSeats)
    {
        _context.Logger().Log($"Booking {numSeats} seats");
        _context.Bus().Emit(new BookingRequested(numSeats));
    }
}